### PR TITLE
Optimization for loading widget settings

### PR DESCRIPTION
**Pull request by scadasystems**
Fixed the refreshing widget when every text input.

### CODE REVIEW

File: lib/src/components/form/form.dart [hunk 1]
```diff
@@ -1444,6 +1444,7 @@ class SubmitButton extends StatefulWidget {
 class _SubmitButtonState extends widgets.State<SubmitButton> {
   Future<bool>? _future;
   FormController? _controller;
+  bool _isSubmitting = false;
```
Comment (sunarya-thito) [inline lib/src/components/form/form.dart:1447]: Can you elaborate on why we need this field here? The FutureBuilder should handle this one already, or am I missing something?

File: lib/src/components/locale/shadcn_localizations.dart [hunk 2]
```diff
@@ -736,3 +711,879 @@ class DefaultShadcnLocalizations extends ShadcnLocalizations {
   @override
   String get colorLightness => 'Lum';
 }
+
```
Comment (sunarya-thito) [inline lib/src/components/locale/shadcn_localizations.dart:714]: I just realized that this file will have thousands of lines when we add a new locale, can we separate this into `korean_shadcn_localizations.dart` and `japanese_shadcn_localizations.dart`?

**Review by sunarya-thito — CHANGES_REQUESTED** [review]

File: lib/src/components/locale/shadcn_localizations.dart (hunk 2 above)
Comment (scadasystems) [reply lib/src/components/locale/shadcn_localizations.dart]: I push the file separately

File: lib/src/components/form/form.dart (hunk 1 above)
Comment (sunarya-thito) [reply lib/src/components/form/form.dart:1447]: I don't think the issue is caused by the submit button, checked it myself yesterday, it was the unnecessary microtask inside the Form component. I pushed a commit that solves this issue, you can rebase it and this change is no longer necessary.

Comment (scadasystems) [reply lib/src/components/form/form.dart:1447]: Okay. Thank you

**Pull request closed without merging**